pages/onboarding/index.tsx

```diff
@@ -1,15 +1,129 @@
-import { Box, Button, Container, Typography } from '@mui/material';
+import { Box, Button, Container, TextField, Typography } from '@mui/material';
+import axios from 'axios';
+import jwt from 'jsonwebtoken';
 import { NextPage } from 'next';
 import Link from 'next/link';
+import { useRouter } from 'next/router';
+import React, { useState } from 'react';
+import Cookies from 'universal-cookie';
+
+import createAxiosInstance from 'client/lib/axios';
+import { UserProfile } from 'server/types/User';
+import { AUTH_COOKIE_NAME } from 'shared/constants';
+
+interface FormData {
+  bio: string;
+  twitter_username: string;
+  linkedin_url: string;
+  github_username: string;
+  website: string;
+}
+
+interface Props {
+  user: UserProfile;
+}
+
+const OnboardingPage: NextPage<Props> = ({ user }: Props) => {
+  const [formData, setFormData] = useState<FormData>({
+    bio: user.bio,
+    twitter_username: user.twitter_username,
+    linkedin_url: user.linkedin_url,
+    github_username: user.github_username,
+    website: user.website,
+  });
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isFormFilled, setIsFormFilled] = useState<boolean>(false);
+
+  const router = useRouter();
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setFormData(prevFormData => ({ ...prevFormData, [name]: value }));
+    setIsFormFilled(true);
+  };
```

Comment:
The way this is implemented means that anytime the user changes the form the state is changed to `true` for setIsFormFilled even if the user removes the text they typed in. The button still becomes enabled so they can send another patch request with the same data they had before even if that data is exact same.

in this example, on the first `/onboarding` load after login, I filled out `bio, twitter username, linkedin url` and then came back to the page and filled out one of the empty fields partially then removed what I typed in. but I was still able to submit the data even though nothing changed
![image](https://user-images.githubusercontent.com/68509315/234382977-0882ac1d-1a8f-4f83-9462-bbdb02db1a73.png)